### PR TITLE
Read file s3 support

### PR DESCRIPTION
**Pull request by luciferlinx101**
### Description

Adding support for S3 in read files

### Related Issues

Cannot read files in S3

### Solution and Design

I have improved read file which can read from both S3 and File based on config.yaml along with first checking input dir and if not found checking in output dir.

### Test Plan

Set storage type S3 and use read tool

### Type of change
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Docs update

### Checklist
- [x] My pull request is atomic and focuses on a single change.
- [x] I have read the contributing guide and my code conforms to the guidelines.
- [x] I have documented my changes clearly and comprehensively.
- [x] I have added the required tests.

### CODE REVIEW

File: superagi/helper/s3_helper.py [hunk 1]
```diff
@@ -35,3 +36,17 @@ def upload_file(self, file, path):
             logger.info("File uploaded to S3 successfully!")
         except:
             raise HTTPException(status_code=500, detail="AWS credentials not found. Check your configuration.")
+
+    @classmethod
+    def get_s3_client(cls):
```
Comment (TransformerOptimus) [inline superagi/helper/s3_helper.py:41]: u can use this method in this class constructor as well.

File: superagi/helper/resource_helper.py [hunk 2]
```diff
@@ -118,6 +119,12 @@ def get_agent_write_resource_path(cls, file_name: str, agent: Agent, agent_execu
         final_path = root_dir + file_name
         return final_path
 
+    @classmethod
+    def check_file_exists_in_s3(cls, file_path):
+        s3_client = S3Helper.get_s3_client()
```
Comment (TransformerOptimus) [inline superagi/helper/resource_helper.py:124]: this logic can move inside s3_helper.

File: superagi/resource_manager/file_manager.py [hunk 3]
```diff
@@ -53,6 +54,14 @@ def write_to_s3(self, file_name, final_path):
                     s3_helper = S3Helper()
                     s3_helper.upload_file(img, path=resource.path)
 
+    def read_from_s3(self, file_path):
+        file_path = "resources" + file_path
+        logger.info(f"Reading file from s3: {file_path}")
+        response = S3Helper.get_s3_client().get_object(Bucket=get_config("BUCKET_NAME"), Key=file_path)
```
Comment (TransformerOptimus) [inline superagi/resource_manager/file_manager.py:60]: use s3 helper as abstraction for all s3 methods. u can give these as object methods

**Review by DMTarmey — COMMENTED** [review]
help

File: superagi/helper/resource_helper.py [hunk 4]
```diff
@@ -128,13 +135,18 @@ def get_agent_read_resource_path(cls, file_name, agent: Agent, agent_execution:
...
         """
-        output_root_dir = ResourceHelper.get_root_output_dir()
         final_path = ResourceHelper.get_root_input_dir() + file_name
         if "{agent_id}" in final_path:
             final_path = ResourceHelper.get_formatted_agent_level_path(
                 agent=agent,
                 path=final_path)
-        if final_path is None or not os.path.exists(final_path):
+        output_root_dir = ResourceHelper.get_root_output_dir()
+        if final_path is None or (StorageType.get_storage_type(get_config("STORAGE_TYPE",
```
Comment (TransformerOptimus) [inline superagi/helper/resource_helper.py:144]: condition is too complex. Simplify it or move it to a method.

File: superagi/helper/s3_helper.py (hunk 1 above)
Comment (luciferlinx101) [reply superagi/helper/s3_helper.py:41]: okay

File: superagi/resource_manager/file_manager.py (hunk 3 above)
Comment (luciferlinx101) [reply superagi/resource_manager/file_manager.py:60]: Done!

**Pull request closed without merging**